### DISCUSSION TITLE
The paramters in main function doesn't work

### DISCUSSION THREAD

**Discussion by nickma1204** [pytest-dev/pytest#10095] [Q&A] [answered]
I'm trying to setup a pytest demo and insert some paramter in main function, however the paratmeres '-s' or test case selected parameters doesn't work, could you please help to find out where the problem is?

```
import pytest


def test_02():
    print("test02")


def test_01():
    print("test01")


if __name__ == '__main__':
    pytest.main(['-s', 'test_case.py::test_01'])
```

We can see from the result ,both of them is excuted and no string is printed out

```
============================= test session starts =============================
collecting ... collected 2 items

test_case.py::test_02 PASSED                                             [ 50%]test02

test_case.py::test_01 PASSED                                             [100%]test01


============================== 2 passed in 0.02s ==============================
```
here is my pytest version info

`platform win32 -- Python 3.10.1, pytest-7.1.2, pluggy-1.0.0`

**Comment by The-Compiler** [comment]
What's the command you're running now?

**Reply by nickma1204** [reply]
I run it in pycharm just click the Run button next to the main function, I wounder why it doesn't work in pycharm

I try the command " pytest test_case.py::test_01 -s" in terminal, it works fine

**Comment by nickma1204** [comment, marked as answer]
Eventually I found the root cause.
when I click the run button next to the main, it run default 'Run python test for test....', this will run all the test under this module and ignore whatever I insert in the pytest.main().
If in menu bar, I click Run->Run->test_case.py to navigate to my test case module, it will run with the args in main.

**Reply by RonnyPfannschmidt** [reply]
To explain this more - pycharm pytest directly runs pytest.main itself

If you want to run specific tests, just click the run button next to the test function definition
